hazelcast/src/main/java/com/hazelcast/internal/partition/impl/InternalPartitionServiceImpl.java

```diff
@@ -1247,22 +1244,57 @@ private void processResults(Collection<Future<PartitionRuntimeState>> futures,
                     if (state.getActiveMigration() != null) {
                         allActiveMigrations.add(state.getActiveMigration());
                     }
-                } catch (TargetNotMemberException e) {
-                    ignore(e);
-                } catch (MemberLeftException e) {
-                    ignore(e);
-                } catch (InterruptedException e) {
-                    currentThread().interrupt();
-                    logger.fine("FetchMostRecentPartitionTableTask is interrupted.");
-                } catch (ExecutionException e) {
-                    Throwable cause = e.getCause();
-                    if (!(cause instanceof TargetNotMemberException) && !(cause instanceof MemberLeftException))  {
-                        logger.warning("Failed to fetch partition table!", e);
-                    }
                 }
             }
         }
 
+        /**
+         * Fetches known partition state from next member and returns null if target member is left
+         * and/or not a member of this cluster anymore.
+         * If future timeouts, then fetch operation is retried until we learn target member's partition state
+         * or it leaves the cluster.
+         */
+        private PartitionRuntimeState collectNextPartitionState(Iterator<Map.Entry<Member, Future<PartitionRuntimeState>>> iter) {
+            Map.Entry<Member, Future<PartitionRuntimeState>> next = iter.next();
+            Member member = next.getKey();
+            Future<PartitionRuntimeState> future = next.getValue();
+            boolean collectedState = true;
+
+            try {
+                PartitionRuntimeState state = future.get(FETCH_PARTITION_STATE_SECONDS, TimeUnit.SECONDS);
+                if (state == null) {
+                    logger.fine("Received NULL partition state from " + member);
+                } else {
+                    logger.fine("Received partition state version: " + state.getVersion() + " from " + member);
+                }
+                return state;
+            } catch (TargetNotMemberException e) {
```

Comment:
Hmm, would it be a good idea to catch all other exception after `TimeoutException` with `EmptyStatement.ignore(e)`? I'm just thinking what would happen, if an unexpected exception would be received here. This just looks a bit more safe to me:
```java
try {
    (...)
} catch (TimeoutException e) {
    collectedState = false;
    // Fetch partition state operation is idempotent.
    // We will retry it until it we learn the partition state or the member leaves the cluster.
    // We can't just rely on invocation retries, because if connection is dropped while
    // our operation is on the wire, invocation won't get any response and will eventually timeout.
    next.setValue(fetchPartitionState(member));
} catch (Exception e) {
    EmptyStatement.ignore(e);
} finally {
    (...)
}
```
I thought about another simplification, but `MemberLeftException extends ExecutionException` prevents that.

Reply:
make sense.